gapis/api/vulkan/display_to_surface.go

```diff
@@ -0,0 +1,106 @@
+// Copyright (C) 2020 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vulkan
+
+import (
+	"context"
+
+	"github.com/google/gapid/gapis/api"
+)
+
+type displayToSurface2 struct {
+	SurfaceTypes map[uint64]uint32
+}
+
+func NewDisplayToSurface2() *displayToSurface2 {
+	return &displayToSurface2{
+		SurfaceTypes: map[uint64]uint32{},
+	}
+}
+
+func (t *displayToSurface2) RequiresAccurateState() bool {
+	return false
+}
+
+func (surfaceTransform *displayToSurface2) BeginTransform(ctx context.Context, inputCommands []api.Cmd, inputState *api.GlobalState) ([]api.Cmd, error) {
+	return inputCommands, nil
+}
+
+func (surfaceTransform *displayToSurface2) EndTransform(ctx context.Context, inputCommands []api.Cmd, inputState *api.GlobalState) ([]api.Cmd, error) {
+	return inputCommands, nil
+}
+
+func (surfaceTransform *displayToSurface2) ClearTransformResources(ctx context.Context) {
+	// Do nothing
+}
+
+func (surfaceTransform *displayToSurface2) TransformCommand(ctx context.Context, id api.CmdID, inputCommands []api.Cmd, inputState *api.GlobalState) ([]api.Cmd, error) {
+	for i, cmd := range inputCommands {
+		modifiedCmd := surfaceTransform.modifySurface(ctx, cmd, inputState)
+		if modifiedCmd != nil {
+			inputCommands[i] = modifiedCmd
+		}
+	}
+
+	return inputCommands, nil
+}
+
+// displayToSurface2 is a transformation that enables rendering during replay to
```

Comment:
The godoc for the type should be on the type, this godoc should describe what the function is doing.